### PR TITLE
net-proxy/mitmproxy: Bump to version 5.2.0

### PR DESCRIPTION
**Pull request by 00-matt**

### CODE REVIEW

File: net-proxy/mitmproxy/mitmproxy-5.2.ebuild [hunk 1]
```diff
@@ -0,0 +1,80 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+DISTUTILS_USE_SETUPTOOLS=rdepend
+PYTHON_COMPAT=( python3_7 python3_8 )
```
Comment (hydrapolic) [inline net-proxy/mitmproxy/mitmproxy-5.2.ebuild:7]: python3_{7,8}

Comment (00-matt) [reply net-proxy/mitmproxy/mitmproxy-5.2.ebuild:7]: Thanks :) updated.

File: net-proxy/mitmproxy/mitmproxy-5.2.ebuild [hunk 2]
```diff
@@ -0,0 +1,80 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+DISTUTILS_USE_SETUPTOOLS=rdepend
+PYTHON_COMPAT=( python3_{7,8} )
```
Comment (thesamesam) [inline net-proxy/mitmproxy/mitmproxy-5.2.ebuild:7]: Some deps are in the way of 3.9, right?

File: net-proxy/mitmproxy/mitmproxy-5.2.ebuild [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+	>=dev-python/ruamel-yaml-0.15[${PYTHON_USEDEP}]
+	>=dev-python/sortedcontainers-1.5.4[${PYTHON_USEDEP}]
+	>=www-servers/tornado-4.3[${PYTHON_USEDEP}]
+	>=dev-python/urwid-2.0.1[${PYTHON_USEDEP}]
+	>=dev-python/wsproto-0.11.0[${PYTHON_USEDEP}]
+	>=dev-python/zstandard-0.11.0[${PYTHON_USEDEP}]
+"
+
+DEPEND="${RDEPEND}
+	dev-python/setuptools[${PYTHON_USEDEP}]
```
Comment (thesamesam) [inline net-proxy/mitmproxy/mitmproxy-5.2.ebuild:47]: Can drop this or specify with the [eclass](https://dev.gentoo.org/~mgorny/python-guide/distutils.html#different-build-system-variations).

File: net-proxy/mitmproxy/mitmproxy-5.2.ebuild [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+	# remove failing test
+	sed -i 's/test_get_version/_&/g' test/mitmproxy/test_version.py || die
+
+	# requires asynctest
+	rm test/mitmproxy/addons/test_readfile.py || die
+
+	distutils-r1_python_prepare_all
+}
+
+python_test() {
```
Comment (thesamesam) [inline net-proxy/mitmproxy/mitmproxy-5.2.ebuild:78]: Unnecessary with distutils_enable_tests.

File: net-proxy/mitmproxy/mitmproxy-5.2.ebuild (hunk 2 above)
Comment (00-matt) [reply net-proxy/mitmproxy/mitmproxy-5.2.ebuild:7]: Upstream also does not support Python 3.9. There is some work in the master branch and in some open pull requests.

Comment (hydrapolic) [reply net-proxy/mitmproxy/mitmproxy-5.2.ebuild:7]: https://github.com/mitmproxy/mitmproxy/pull/4174

Comment (thesamesam) [reply net-proxy/mitmproxy/mitmproxy-5.2.ebuild:7]: Thanks, no worries. Re what you asked on IRC (should you add patches), the answer is yes, if they're not too invasive, but only if they're upstreamed ideally. Let's just remember to come back to it!

**Pull request closed without merging**